### DISCUSSION TITLE
Startup

### DISCUSSION THREAD

**Discussion by Jhl44** [joeyagreco/leeger#83] [General]
Hey, I have been trying to use your code to create something for my espn fantasy league and am confused on how to get the process started to load all the things. I am a beginner to coding using github but I installed the code using python and have the files but not too sure what to do after having this. If you could give me a more detailed instruction on how to proceed that would be awesome, thank you

**Comment by joeyagreco** [comment]
Thanks for your interest in this library!

I've added a [Quickstart Guide](https://github.com/joeyagreco/leeger#quickstart-guide) to make it easier to use this library.

Please let me know if you have any other questions/issues.

**Reply by Jhl44** [reply]
In step 5 of the instructions, this error is coming up, any idea what I am doing wrong in it?


Traceback (most recent call last):
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\leeger\league_loader\ESPNLeagueLoader.py", line 36, in __init__
    int(leagueId)
ValueError: invalid literal for int() with base 10: 'THE Fantasy League'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Downloads\my_file.py", line 17, in <module>
    espnLeagueLoader = ESPNLeagueLoader("THE Fantasy League", [2020, 2022], espnS2=espnS2, swid=swid)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\leeger\league_loader\ESPNLeagueLoader.py", line 38, in __init__
    raise ValueError(f"League ID '{leagueId}' could not be turned into an int.")
ValueError: League ID 'THE Fantasy League' could not be turned into an int.

**Reply by joeyagreco** [reply]
League Loaders take a League ID (not a league name).

In the setup documentation for your specific site, it will tell you where you can find your league ID.

**Reply by Jhl44** [reply]
*facepalm* .... now where in the leagueloader code do I add the string of code to define the multiple names within the same team across different years? I saw the example posted in the code, but where do I go about adding it.
Thank you for all this help too, appreciate it a lot

**Reply by joeyagreco** [reply]
You can do something like [this](https://github.com/joeyagreco/leeger/blob/main/example/league_loader/nameChanges.py) and pass in the dictionary of owner names and aliases directly into the league loader instantiation

**Reply by Jhl44** [reply]
its weird, I have tried many combinations of that code in the league loader and have been trying to play around with it but it seems to always give a syntax error no matter what I do when I try to add an alias for that one manager

**Reply by joeyagreco** [reply]
Could you send the full script that you are running as well as the error logs please?

**Reply by Jhl44** [reply]
I have added the most recent code files of "league loader" and "my_file" under the leeger/league_loader section of my repositories , they are set up to the most recent point of what I was attempting to work through the errors I was receiving, however different combinations of it were tried

**Reply by joeyagreco** [reply]
It's hard to debug what the issue is without seeing the code that you're running but I've attached a script that may help.

```python
from leeger.league_loader import ESPNLeagueLoader, SleeperLeagueLoader, MyFantasyLeagueLeagueLoader
from leeger.model.league import League
from leeger.util.excel import leagueToExcel

if __name__ == "__main__":
    # Here, we want the owner name for John Smith to be "John Smith" and the name for Jane Smith to be "Jane Smith"
    # The keys of the dictionary are the names that you want the Owners to be set to
    # and the list of values in the dictionary are the names that will appear in the fantasy league sites
    # This will map the names "John Smith" and "Johnny Smith" to the name "John Smith"
    # This will also map the names "Jane Smith" and "Janey" to the name "Jane Smith"
    # We will pass this dictionary into each league loader constructor
    owner_names_and_aliases = {"John Smith": ["John Smith", "Johnny Smith"],
                               "Jane Smith": ["Jane Smith", "Janey"]}
    # 2019
    espnS2 = "foo"
    swid = "bar"
    espn_league_loader = ESPNLeagueLoader("123456",
                                          [2019],
                                          espnS2=espnS2,
                                          swid=swid,
                                          ownerNamesAndAliases=owner_names_and_aliases)
    espn_league: League = espn_league_loader.loadLeague()

    # 2020
    sleeper_league_loader = SleeperLeagueLoader("654321",
                                                [2020],
                                                ownerNamesAndAliases=owner_names_and_aliases)
    sleeper_league: League = sleeper_league_loader.loadLeague()

    # 2021-2022
    MFL_USERNAME = "myUsername"
    MFL_PASSWORD = "myPassword"
    MFL_USER_AGENT_NAME = "myUserAgentName"
    LEAGUE_ID = "10101010101"

    myFantasyLeagueLoader = MyFantasyLeagueLeagueLoader(LEAGUE_ID,
                                                        [2021, 2022],
                                                        mflUsername=MFL_USERNAME,
                                                        mflPassword=MFL_PASSWORD,
                                                        mflUserAgentName=MFL_USER_AGENT_NAME,
                                                        ownerNamesAndAliases=owner_names_and_aliases)
    mfl_league: League = myFantasyLeagueLoader.loadLeague()

    # combine leagues
    # skip this step if you're pulling from the same league loader for all leagues
    # in this example, we are pulling from ESPN, Sleeper, and MyFantasyLeague, so we need to combine them into 1 League object
    combined_league = espn_league + sleeper_league + mfl_league

    leagueToExcel(combined_league, "C:\\Users\\my_stats.xlsx", overwrite=True)


```

**Reply by joeyagreco** [reply]
Also I'll note that for this you are NOT editing the leeger code.

You are creating your own separate script locally and using the leeger code via importing the library.

**Reply by Jhl44** [reply]
This is the latest error code I am receiving

Traceback (most recent call last):
  File "C:\Users\Downloads\my_file.py", line 1, in <module>
    from leeger.league_loader import ESPNLeagueLoader
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\leeger\league_loader\__init__.py", line 1, in <module>
    from .ESPNLeagueLoader import ESPNLeagueLoader
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\leeger\league_loader\ESPNLeagueLoader.py", line 6, in <module>
    from leeger.league_loader.LeagueLoader import LeagueLoader
  File "C:\Users\AppData\Local\Programs\Python\Python311\Lib\site-packages\leeger\league_loader\LeagueLoader.py", line 28
    ownerNamesAndAliases = ("ryan o": ["Brandon Todd", "ryan o"],
                                                             ^
SyntaxError: invalid syntax

**Reply by Jhl44** [reply]
the arrow is pointing to the ":" part after listing the name

**Reply by joeyagreco** [reply]
ownerNamesAndAliases should be a Python dictionary.

`owner_names_and_aliases = {"John Smith": ["John Smith", "Johnny Smith"],
                               "Jane Smith": ["Jane Smith", "Janey"]}`

Use "{}" not "()"

**Reply by Jhl44** [reply]
in my code itself I am using those and it is still showing up as that error. I am not sure why it is showing up different in the error but in the code it is the right brackets

**Reply by joeyagreco** [reply]
If you send your full script I can debug it. 

If you'd rather not post it here you can send it to my email: joeyagreco@gmail.com.

Otherwise I'm happy to hop on a call and work through this.
